src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/PolymorphicTypeResolver.cs

```diff
@@ -15,18 +15,16 @@ namespace System.Text.Json.Serialization.Metadata
     /// </summary>
     internal sealed class PolymorphicTypeResolver
     {
-        private readonly JsonTypeInfo _declaringTypeInfo;
         private readonly ConcurrentDictionary<Type, DerivedJsonTypeInfo?> _typeToDiscriminatorId = new();
         private readonly Dictionary<object, DerivedJsonTypeInfo>? _discriminatorIdtoType;
+        private readonly JsonSerializerOptions _options;
 
-        public PolymorphicTypeResolver(JsonTypeInfo jsonTypeInfo)
+        public PolymorphicTypeResolver(JsonSerializerOptions options, JsonPolymorphismOptions polymorphismOptions, Type baseType, bool converterCanHaveMetadata)
```

Comment:
Can the resolver now be used in cases where a base `JsonTypeInfo` is not available or applicable?

Reply:
Yes, I mostly did this change to ensure Configure ordering was less error-prone and correct (semi-configured JsonTypeInfo could cause unexpected behavior while those values directly won't change after initialized)